### DISCUSSION TITLE
Incoming Network Connection on macOS

### DISCUSSION THREAD

**Discussion by dogrady2** [idaholab/moose#16876] [Q&A General] [answered]
Hi All,

Whenever I run sam on my macOS Catalina laptop I get the following pop-up
 
![Screen Shot 2020-11-20 at 10 47 38 AM (1)](https://user-images.githubusercontent.com/35669108/106785649-3c7a6c80-6613-11eb-9eda-fe3a12dab52d.png)

The pop-up occurs regardless of the model that is being run, it also occurs when running --dump. Other sam developer/users do not report getting the pop-up. I have not been able to traceback what is causing the network connection. I am not sure if the cause of the pop-up is in the compilation environment, sam level or moose level. Does anyone have any suggestions for disabling the pop-up? 

Thanks,
Dan O'Grady

**Comment by permcody** [comment, marked as answer]
Hi Dan, To the best of my knowledge there are two ways around this:
 - Turn off your firewall - easy to do, but maybe not the best way since it could open you up to potential security issues.
 - Sign your binary with an Apple Developer certificate - This of course means you have to register and become an Apple Developer. If you have a signed binary, it won't be flagged by Mac OS's firewall. The issue with this is that you have to do this every time you build. We already have a line in our Makefile that makes MacOS binaries debuggable. It would be easy to add a line if we could do this generically for everyone, but I'm not aware of a solution that will work for everyone since you (as the end-developer) are creating the binary as you so you have to be the one that signs it each time it is created.

**Reply by travismui** [reply]
Hi @permcody, thanks for your notes on this. Just to add onto @dogrady2, I've also started experiencing this pop-up issue this week (I think he noticed this back in November 2020), so I've tried some troubleshooting.
- This seems to affect any Moose app, I noticed this with Bison and also `ex01-opt` in the Moose examples, so I'm curious if others in the Moose community have noticed the pop-ups recently.
- We suspect it's rooted in the latest macOS Security Update 2021-001 from Feb 1 (see [Apple](https://support.apple.com/en-us/HT212147)) that has triggered this 'bug'.
- The incessant pop-ups occur only when the macOS firewall is on and the first time after you build your executable. After a restart, the pop-ups don't show up but they do after you re-build.

A couple details on what I checked to diagnose the error:
- Our work machines are on macOS Catalina, but I reproduced this issue on Mojave too (see below)
- Turning off the firewall fixes the issue– but this may not be an option for some users.
- We are not enrolled in the Apple Developer program so I don't think we can sign with a certificate. It does seem that Sam is not signed when I checked with `codesign`.
- On a personal machine using Mojave, before applying the Security Update I did not experience the pop-up dialogue. After applying the Security Update, I get the pop-up issue as described above. Unfortunately, I don't know of any easy way to unroll a macOS Security Update without restoring to a backup image.
